### PR TITLE
Watch, recompile and reload view changes

### PR DESCRIPTION
**Pull request by baelter**
### WHAT is this pull request doing?
To enable fast iteration on UI development this PR adds a couple of things:

1. make targets and associated files to compile ECR views separately and serve them from the static/ folder.
2. make target to watch ECR files and recompile them when they change.
3. livereload client and server to refresh the browser when views a recompiled.

### HOW can this pull request be tested?
Run two terminals with
`bin/lavinmq`
and
`make dev-ui`
Open a browser and make some changes to ECR files and behold.

### CODE REVIEW

File: views/vhost.ecr [hunk 1]
```diff
@@ -110,6 +110,6 @@
         </form>
       </section>
     </main>
-    <% render "footer" %>
+    <% render "partials/footer" %>
   </body>
-</html>
+</etml>
```
Comment (carlhoerberg) [inline views/vhost.ecr:115]: Mistake?

File: views/vhost.ecr [hunk 2]
```diff
@@ -1,12 +1,12 @@
 <%- pagename = "Virtual host" -%>
-<!DOCTYPE html>
+<!DOCTYPE >
```
Comment (carlhoerberg) [inline views/vhost.ecr:2]: ?

File: views/partials/head.ecr [hunk 3]
```diff
@@ -10,3 +10,6 @@
 <link rel="apple-touch-icon" sizes="180x180" href="img/favicon.png">
 <meta name="viewport" content="width=device-width, initial-scale=1.0">
 <meta name="google" content="notranslate">
+<%- {% unless flag?(:release) %} -%>
+  <script src="http://localhost:35729/livereload.js"></script>
```
Comment (carlhoerberg) [inline views/partials/head.ecr:14]: Can the liveview server be started and integrated in Lavinmqs http server?

File: shard.lock [hunk 4]
```diff
@@ -12,6 +12,14 @@ shards:
...
     version: 1.2.7
 
+  inotify:
+    git: https://github.com/petoem/inotify.cr.git
+    version: 1.0.3
+
+  live_reload:
+    git: https://github.com/bcardiff/live_reload.cr.git
+    version: 0.1.0
+
```
Comment (carlhoerberg) [inline shard.lock:22]: Could be development dependencies?

Comment (baelter) [reply shard.lock:22]: They are?

File: views/partials/head.ecr (hunk 3 above)
Comment (baelter) [reply views/partials/head.ecr:14]: Yes

Comment (baelter) [reply views/partials/head.ecr:14]: ... but, I don't think it's worth the effort? You could perhaps set another http server, setup the websocket handler, keep tabs of connection and handle the js yourself. https://github.com/bcardiff/live_reload.cr/blob/master/src/live_reload.cr#L140

But running a separate server like this seems to the standard thing to do https://github.com/livereload/livereload-js?tab=readme-ov-file#what-is-livereload

**Comment by spuun** [comment]
```
 6 | watcher = Inotify.watch dir do |event|
                       ^----
Error: undefined method 'watch' for Inotify:Module
```

Inotify only supports linux:
https://github.com/petoem/inotify.cr/blob/master/src/inotify.cr#L3

**Review by spuun — APPROVED** [review]

**Comment by viktorerlingsson** [comment]
Shouldn't live_reload be a `development_dependency`?

**Comment by baelter** [comment]
> Shouldn't live_reload be a `development_dependency`?

No, I removed the dependency in running live_reload in crystal. https://github.com/cloudamqp/lavinmq/pull/802#discussion_r1795464183

**Pull request merged**